grafana-plugin/src/containers/UserSettings/parts/tabs/PersonalWebhookInfo/PersonalWebhookInfo.tsx

```diff
@@ -1,27 +1,202 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
-import { Stack } from '@grafana/ui';
+import { css } from '@emotion/css';
+import { LoadingPlaceholder, Button, Select, Stack, Field, TextArea } from '@grafana/ui';
 import { StackSize } from 'helpers/consts';
 import { observer } from 'mobx-react';
+import { Controller, useForm } from 'react-hook-form';
 
-import { Block } from 'components/GBlock/Block';
+import { PluginLink } from 'components/PluginLink/PluginLink';
 import { Text } from 'components/Text/Text';
+import { WithConfirm } from 'components/WithConfirm/WithConfirm';
+import { ActionKey } from 'models/loader/action-keys';
+import { WebhookTriggerType } from 'models/outgoing_webhook/outgoing_webhook.types'
+import { useStore } from 'state/useStore';
+
+interface FormFields {
+  webhook: string;
+  context: string;
+}
+
+function useLoadingWebhooks() {
+  const { loaderStore } = useStore();
+  const [isLoading, setIsLoading] = React.useState(true)
+  const isLoadingWebhooks = loaderStore.isLoading(ActionKey.FETCH_WEBHOOKS)
+  const [hasRegisteredLoadingWebhooks, setHasRegisteredLoadingWebhooks] = useState(false);
+
+  useEffect(() => {
+    if (isLoadingWebhooks === true) {
+      setHasRegisteredLoadingWebhooks(true);
+    }
+  }, [isLoadingWebhooks]);
+
+  useEffect(() => {
+    if (!isLoadingWebhooks && hasRegisteredLoadingWebhooks) {
+      setIsLoading(false);
+    }
+  }, [isLoadingWebhooks])
+
+  return isLoading;
+}
```

Comment:
`loaderStore.isLoading(ActionKey.FETCH_WEBHOOKS)` returns false on the first couple of renders which results in a flicker as the form is rendered briefly before showing the loading placeholder. I wrote this hook to prevent that but it feels like I shouldn't have to so perhaps I'm missing something here.